providers/implementations/keymgmt/ecx_kmgmt.c

```diff
@@ -510,6 +510,7 @@ static void *ecx_gen_init(void *provctx, int selection,
 #endif
     }
     if (!ecx_gen_set_params(gctx, params)) {
+        OPENSSL_free(gctx->propq);
```

Comment:
It is a memory leak path,
I think this should just call ecx_gen_cleanup(gctx) instead.
If someone adds something to the end of the set_params() we will have the same problem again in the future. Which is why this became a problem in the first place..

Reply:
Right, changed.